### DISCUSSION TITLE
how to enable the debug log for lxqt-session and pcmanfm-qt

### DISCUSSION THREAD

**Discussion by kalgnana** [lxqt/lxqt#2452] [General]
Hi Team, 

       We are working with pcmanfm-qt (0.17.0) for few customisation to bring functionalities for handling the virtual drive handling in file manager. As part of that, we have successfully complied the file manager source code, and also able to run the compiled version. but not able to enable the debug logs. 

We have enabled few debug prints 'on_actionXXXX_triggered' callback, but not getting on console. 
                 on_actionXXXX_triggered  (I think this callback called from QXcbEventQueue thread)

Could you help us on 'How to enable the logging for Lxqt components in general, and specifically for pcmanfm-qt' 

Thanks & Regards,
Kalyan-G.

**Comment by tsujan** [comment]
I'm not sure if I've understood your question, but pcmanfm-qt doesn't log anything (why should it?). There are only a few debug messages created by libfm-qt, which are redundant and might be removed. If you want it to create a log file, you'll need to add your codes to it and/or to libfm-qt.

> We are working with pcmanfm-qt (0.17.0)

0.17.0 is an ancient version, although this isn't related to your question.

**Comment by kalgnana** [comment]
We are trying to understand the pcmanfm-qt architecture and code flow by enable more custom debug logs in the code. As part of that, we have enabled log around 'on_xxxx_triggered' ('mainwindow.cpp') functions, and loaded that compiled binary. We were expecting the logs in console, but it is not throw any logs in consoles.  Copied one of the custom logs sample here. 

`void MainWindow::on_actionAbout_triggered() {
    // the about dialog
    class AboutDialog : public QDialog {
            ...
            qDebug("on_actionAbout_triggered: %d", 100);
            std::cout << "Hello, World!" << std::endl;
`

We have modified the desktop entry file also to redirect the logs into a file.
 
`/usr/local/bin/pcmanfm-qt --desktop --profile=lxqt 2> /tmp/log.txt`

**Reply by tsujan** [reply]
> We were expecting the logs in console...

What you did should work.

Check if your distro has changed Qt's logging level somehow. I don't know how it can be changed globally (with compilation and/or a root file?), but the user can change itl in `~/.config/QtProject/qtlogging.ini` by adding rules like

```
[Rules]
*.warning=false
```

Maybe an LXQt member with a knowledge of Qt compilation options could give you better info.

Anyway, what *I* can tell is that the essential functionalities of pcmanfm-qt are provided by libfm-qt. So, you might want to do the same thing with libfm-qt after your problem is solved.

Lastly, I absolutely can't recommend V0.17.0. Use the latest release instead.
